Makefile

```diff
@@ -0,0 +1,6 @@
+all            :; dapp --use solc:0.5.12 build
```

Comment:
Can we use a later solidity version with optimizations?

Reply:
I am not particularly tempted to enable optimisations, given how infrequently this code will be called. Are there other reasons that you have in mind for bumping solidity version?

Reply:
I'm not really sure how frequently this will be called. Just suggesting optimizations because we can usually get an immediate ~20% gas efficiency out of it. Newer solc just for bugfixes and other efficiencies. I'm fine with 0.5.12 if you want to stick with it though.

Reply:
Unless anyone has strong feelings otherwise, I'd prefer to say on 0.5.12 here and not enable optimisations. The reason for not enabling optimisations is that all of the code will be called infrequently, either by governance or by a single operator, so the reward for the risk/extra bytecode size is low. The reason to stay on 0.5.12 is that there do not appear to be [known bugs](https://docs.soliditylang.org/en/v0.8.0/bugs.html) affecting it which would be relevant here, so again it seems to be lower risk to stick with it.